src/main/java/ru/hogwarts/school/service/FacultyService.java

```diff
@@ -1,50 +1,56 @@
 package ru.hogwarts.school.service;
 
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.Objects;
+import java.util.*;
 
 import org.springframework.stereotype.Service;
 import ru.hogwarts.school.model.Faculty;
-
-
+import ru.hogwarts.school.model.Student;
+import ru.hogwarts.school.repositories.FacultyRepository;
+//@AllArgsConstructor
 @Service
 public class FacultyService {
 
-    private final HashMap<Long, Faculty> faculties = new HashMap<>();
-    private long count = 0;
+    private final FacultyRepository facultyRepository;
+    private final StudentService studentService;
 
-    public Faculty addFaculty(Faculty faculty) {
-        faculty.setId(count++);
-        faculties.put(faculty.getId(), faculty);
-        return faculty;
+    public FacultyService(FacultyRepository facultyRepository, StudentService studentService) {
+        this.facultyRepository = facultyRepository;
+        this.studentService = studentService;
     }
 
     public Faculty findFaculty(long id) {
-        return faculties.get(id);
+        return facultyRepository.getById(id);
```

Comment:
Поиск по id написан неправильно:
```suggestion
        return facultyRepository.findById(id).orElseThrow();
```